android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt

```diff
@@ -585,21 +590,46 @@ class Device(
             BluetoothGattDescriptor.DISABLE_NOTIFICATION_VALUE
         }
 
+        val bondReceiver = object : BroadcastReceiver() {
+            override fun onReceive(ctx: Context, intent: Intent) {
+                if (intent.action == BluetoothDevice.ACTION_BOND_STATE_CHANGED) {
+                    val prev = intent.getIntExtra(BluetoothDevice.EXTRA_PREVIOUS_BOND_STATE, -1)
+                    val state = intent.getIntExtra(BluetoothDevice.EXTRA_BOND_STATE, -1)
+                    if (prev == BluetoothDevice.BOND_BONDING && state == BluetoothDevice.BOND_NONE) {
+                        ctx.unregisterReceiver(this)
+                        reject(key, "Pairing request was cancelled by the user.")
+                    }
+                }
+            }
+        }
+        bondReceiverMap[key] = bondReceiver
+        context.registerReceiver(
+            bondReceiver,
+            IntentFilter(BluetoothDevice.ACTION_BOND_STATE_CHANGED)
+        )
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             val statusCode = bluetoothGatt?.writeDescriptor(descriptor, value)
             if (statusCode != BluetoothStatusCodes.SUCCESS) {
+                bondReceiverMap.remove(key)?.let {
+                    context.unregisterReceiver(it)
+                }
                 reject(key, "Setting notification failed with status code $statusCode.")
                 return
             }
         } else {
             descriptor.value = value
             val resultDesc = bluetoothGatt?.writeDescriptor(descriptor)
             if (resultDesc != true) {
+                bondReceiverMap.remove(key)?.let {
+                    context.unregisterReceiver(it)
+                }
                 reject(key, "Setting notification failed.")
                 return
             }
 
         }
+        setTimeout(key, "Set notifications timeout.", timeout)
```

Comment:
```suggestion
        setTimeout(key, "Setting notification timeout.", timeout)
```

to be consistent to other messages

Reply:
I changed that. thx.